cf_xarray/accessor.py

```diff
@@ -436,7 +436,7 @@ def wrapper(obj: Union[DataArray, Dataset], key: str):
     "labels": (_get_indexes,),  # drop_sel
     "coords": (_get_dims,),  # interp
     "indexers": (_get_dims,),  # sel, isel, reindex
-    # "indexes": (_get_axis_coord,),  # set_index
+    "indexes": (_get_dims,),  # set_index TODO: cf_xarray decodes keys, not values
```

Comment:
`.set_index(longitude="lon")` would work, `.set_index(longitude="longitude")` does not work.
I should wrap it in `_single`

Reply:
lets save this for later. this might actually need a reimplementation. We want to change the key with `_get_dims` and the value with `_get_all`?

Reply:
Yes exactly, not sure whether that's something easy to do. I'll add `_single` but will comment everything out.